internal/project/js/build.go

```diff
@@ -71,6 +71,8 @@ func (build *Build) CompileDockerfile(output *dockerfile.Output) error {
 		Description(fmt.Sprintf("builds %s", build.Name())).
 		From("js").
 		Step(step.Arg(nodeBuildArgsVarName)).
+		Step(step.Copy("/", "/generated").From("generate-files")).
+		Step(step.Script(fmt.Sprintf("cp -rf /generated/%s/. ./ || true && rm -rf /generated", build.Name()))).
```

Comment:
this doesn't feel like Dockerfile way to do things ;) why so many copying in the first place?

Reply:
I am trying to copy `/generated/frontend/*` into the frontend dir before doing `npm run build`, because the build might require (in our case, it does) the files to be there. The `|| true` and so on are to make it not fail if there is nothing generated to `frontend/*`, in other words if that dir does not exist.

Reply:
If we used `step.Copy`, we would need it to generate something like `COPY --from=generate-files /frontend/ ./`. This fails if `frontend` directory is not there. What I did with the script is to make it not fail. The intention is "if the `/frontend/` exists on `generate-files`, bring its contents into this directory".

Reply:
> why can't we extend the frontend Generate step to copy files in without proto compliation?

You mean get it generated by `make generate-frontend` and not `make generate`? Because what I understood was to make it work with `make generate`.

And to make it work with `make generate`, the "generated" files has to be copied in before `npm run build` is run.
This is because we have the dependency chain:
`generate` depends on `proto-compile` (because `COPY --from=proto-compile`)
`proto-compile` depends on `tools` (because `FROM tools AS`)
`tools` depend on `frontend` (because `COPY --from=frontend`)
`frontend` depends on the files we are going to generate using `common.Generate` to be there (in our use case)

Therefore I copy the files over in the `frontend` target, before `npm run build`.

Reply:
All I'm saying is KISS. 

There's no big difference between a `.proto` file which is downloaded and compiled to the `.pb.ts` vs. the `.json` file you want to download (and skip compiling). 

All these manual `cp` steps are evil, as they don't allow buildkit to cache properly.